### DISCUSSION TITLE
About the voxel size of fdk reconstruction

### DISCUSSION THREAD

**Discussion by computerAItest** [astra-toolbox/astra-toolbox#401] [Q&A] [answered]
If I want to rebuild with FDK, the original image size is 512*512*120, the voxel size is 0.97*0.97*2.5, the voxel size is not the same on x, y, z,
How to adjust in astra_create_proj_geom? Do you need to keep the image spacing consistent and rebuild after resample?

**Comment by askorikov** [comment, marked as answer]
Hi, using anisotropic voxels is not currently possible with FDK algorithm, but you can indeed try resampling

**Reply by stebaski** [reply]
Hi askorikov, which algo supports anisotropic voxels? Why it is not implemented in FDK? Is it because it's equivalent to doing resampling on the slices after the reconstruction? Just guessing, but maybe with iterative algo (such as SIRT) this is not equivalent... Just curious, thanks!

**Reply by askorikov** [reply]
Hi! For FDK, anisotropic voxels require more complex handling of weighting, which wasn't implemented yet. On the other hand, all iterative algorithms, as well as forward- and backprojection implementations do support anisotropic voxels I believe.

**Comment by orshaf** [comment]
And what about fp3d with not isotropic volumes?
Can I insert the length of the pixels?

**Comment by askorikov** [comment]
You can specify physical size of the reconstructed volume like `vol_geom = astra_create_vol_geom(rows, cols, slices, min_x, max_x, min_y, max_y, min_z, max_z);`, which will define the voxel size along each dimension.

**Reply by orshaf** [reply]
I tried,
But then when I try the create_sino3d_gpu() It asks me to insert as arguments also the volume itself (as numpy array), and then when the dimensions of the numpy array is different then the vol_geom it raises an error.
Do you have a solution?
Thanks!

**Reply by askorikov** [reply]
Well, the dimensions of the numpy array should indeed be the same as in geometry, .i.e it should have `[slices, rows, cols]` shape (that is a convention).

**Reply by orshaf** [reply]
thank you,
I will give details of my problem in a new discussion issue
